servers/quarkus-catalog/src/main/java/org/projectnessie/server/catalog/LakehouseConfigManagementImpl.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.catalog;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.UUID;
+import org.projectnessie.catalog.files.config.ImmutableAdlsOptions;
+import org.projectnessie.catalog.files.config.ImmutableGcsOptions;
+import org.projectnessie.catalog.files.config.ImmutableS3Options;
+import org.projectnessie.catalog.service.api.LakehouseConfigManagement;
+import org.projectnessie.catalog.service.config.ImmutableCatalogConfig;
+import org.projectnessie.catalog.service.config.ImmutableLakehouseConfig;
+import org.projectnessie.catalog.service.config.LakehouseConfig;
+import org.projectnessie.catalog.service.objtypes.ImmutableLakehouseConfigObj;
+import org.projectnessie.catalog.service.objtypes.LakehouseConfigObj;
+import org.projectnessie.versioned.storage.common.exceptions.ObjNotFoundException;
+import org.projectnessie.versioned.storage.common.exceptions.ObjTooLargeException;
+import org.projectnessie.versioned.storage.common.persist.Persist;
+
+public class LakehouseConfigManagementImpl implements LakehouseConfigManagement {
+  private final Persist persist;
+  private final boolean dynamic;
+  private final LakehouseConfig staticLakehouseConfig;
+
+  public LakehouseConfigManagementImpl(
+      Persist persist, boolean dynamic, LakehouseConfig staticLakehouseConfig) {
+    this.persist = persist;
+    this.dynamic = dynamic;
+    this.staticLakehouseConfig = staticLakehouseConfig;
```

Comment:
nit: `staticLakehouseConfig` is not need when `dynamic == true`... I think we can simplify the provider that creates this object by not constructing the static config in that case.